src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -209,63 +248,54 @@ private static boolean longestCommonSubsequenceCheck(String one, String two) {
    *      Similarity</a>
    */
   private static boolean jaroWinklerSimilarityCheck(String one, String two) {
-    return JWS.apply(one, two) > METRIC_THRESHOLD;
+    return JWS.apply(one, two) > JWS_METRIC_THRESHOLD;
   }
 
   /**
-   * Checks if one of the cpeXXUri contains the project's name.
+   * Checks if one of the cpeXXUri contains the project's and organization's names. If the check is
+   * a success, it returns true immediately. If it could not match then check if the list of
+   * references contains any information.
    * 
    * @param cpeMatch of type {@link CpeMatch}.
+   * @param referenceMatch Indicates if there is a reference URL similar to project URL.
+   * @param project of type {@link GitHubProject}.
    * @return True if one of the cpeXXUri contains the project's name. Otherwise False.
    */
-  private boolean cpeUriCheck(CpeMatch cpeMatch) {
-    return match(productFrom23Uri(cpeMatch.getCpe23Uri()))
-        || match(productFrom22Uri(cpeMatch.getCpe22Uri()));
-  }
-
-  /**
-   * Returns product name from {@link CpeMatch#getCpe23Uri()}.
-   */
-  private static String productFrom23Uri(String cpe23Uri) {
-    return stringAtIndex(cpe23Uri, ":", 4);
-  }
+  private static boolean projectCheck(CpeMatch cpeMatch, boolean referenceMatch,
+      GitHubProject project) {
+    CpeUri cpeUri = cpeMatch.getCpeUri();
+
+    boolean productMatch = match(cpeUri.getProduct(), project.name());
+    boolean vendorMatch = match(cpeUri.getVendor(), project.organization().name());
+    // First check if the product and vendor match.
+    if (productMatch && vendorMatch) {
+      return true;
+    }
 
-  /**
-   * Returns product name from {@link CpeMatch#getCpe22Uri()}.
-   */
-  private static String productFrom22Uri(String cpe22Uri) {
-    return stringAtIndex(cpe22Uri, ":", 3);
-  }
+    // Second check if just a reference URL to project exists.
+    if (!productMatch && !vendorMatch && referenceMatch) {
```

Comment:
This may work fine with the set of test projects but I am not sure if it works fine for many other projects. I suspect that could result to many false-positives.

If you still think this is a reliable check, then we can leave it as-is for now. If you'd like to keep it as-is, then you can check if references match much earlier in the line 130.

Reply:
I believe it is a reliable check for now. I tested with multiple scenarios. It always resulted in a wrong outcome. I had this scenario while writing this:
- I wanted vendorMatch to be considered only when productMatch is true. Otherwise not at all.
- Similarly wanted to consider referenceMatch only when both productMatch and vendorMatch are false. 
- If nothing then return whatever the result of productMatch. 

At least for now this check gives the best result.

Reply:
> I wanted vendorMatch to be considered only when productMatch is true. Otherwise not at all.

Currently if `productMatch` is true, then the method always returns `true` no matter what `vendorMatch` is.

> Similarly wanted to consider referenceMatch only when both productMatch and vendorMatch are false.

Why both `productMatch` and `vendorMatch` have to be `false`?

If both vendor and reference match, then the result will be `false`. But if reference match but vendor doesn't, then result will be `true`. I am not sure about that.

Could you please explain? It would be good to put the explanation to a comment in the code if we leave it as-is.